bingo/evaluation/fitness_function.py

```diff
@@ -98,57 +98,30 @@ def __call__(self, individual):
         fitness_vector = self.evaluate_fitness_vector(individual)
         return self._metric(fitness_vector)
 
-    def get_gradient(self, individual):
-        """Vector based fitness gradient
-
-        Get the gradient of this function with respect to an individual's constants.
-
-        Parameters
-        ----------
-        individual : chromosomes
-            individual for which the gradient will be calculated for
-
-        Returns
-        -------
-        gradient :
-            the gradient of this function with respect to each of the individual's constants
-        """
-        fitness_vector = self.evaluate_fitness_vector(individual)
-        fitness_derivatives = self.get_jacobian(individual).transpose()
-
-        gradient = np.zeros(len(fitness_derivatives))
-        for i in range(len(fitness_derivatives)):
-            gradient[i] = self._metric_derivative(fitness_vector, fitness_derivatives[i])
-        return gradient
-
     @abstractmethod
     def evaluate_fitness_vector(self, individual):
         raise NotImplementedError
 
-    @abstractmethod
-    def get_jacobian(self, individual):
-        raise NotImplementedError
-
     @staticmethod
     def _mean_absolute_error(vector):
         return np.mean(np.abs(vector))
 
     @staticmethod
     def _mean_absolute_error_derivative(fitness_vector, fitness_partials):
-        return np.mean(np.sign(fitness_vector) * fitness_partials)
+        return np.mean(np.sign(fitness_vector) * fitness_partials, axis=1)
```

Comment:
lets move the _metric_derivative functionality to the VectorGradientMixin since I don't remember it being used outside that class

Reply:
sounds good, I implemented this in the most recent changes!